modules/currency-l1/src/main/scala/org/tessellation/currency/l1/http/DataApplicationRoutes.scala

```diff
@@ -51,8 +52,10 @@ final case class DataApplicationRoutes[F[_]: Async: Hasher: SecurityProvider: L1
           _.toHashedWithSignatureCheck[F](dataApplication.serializeUpdate _).flatMap {
             case Left(_) => InternalServerError(InvalidSignature.toApplicationError)
             case Right(hashed) =>
-              dataApplication
-                .validateUpdate(hashed.signed.value)
+              OptionT(lastGlobalSnapshotStorage.getOrdinal)
+                .getOrRaise(new IllegalStateException("Global SnapshotOrdinal unavailable"))
+                .map(dataApplication.validateFee)
+                .flatMap(f => (f(hashed.signed), dataApplication.validateUpdate(hashed.signed.value)).mapN(_ |+| _))
```

Comment:
isn't it equivalent to:
```
.flatMap(updates => (dataApplication.validateFee(updates), dataApplication.validateUpdate(hashed.signed.value)).mapN(_ |+| _))
```
?

Reply:
@marcinwadon No it isn't. I refactored it to a single flatMap.